study-frontend/src/Pages/FavoriteTeachersPage.tsx

```diff
@@ -1,23 +1,152 @@
 import React, { useState } from 'react'
 import { Menu } from '../Components/Menu'
 import ClosedMenu from '../Components/Header';
+import { CiSearch } from "react-icons/ci";
+import FilterDropDown from '../Components/FilterDropDown';
+
+
+interface MenuPage{
+    name: string,
+    rout: string, 
+    style: string,
+
+  }
 
 const FavoriteTeachersPage = () => {
     const [clickedFa, setClickedFa] = useState(false);
-
+    
     const handleFaBar = () => setClickedFa(!clickedFa) 
+   
+    const [viewportWidth, setViewportWidth] = useState(window.innerWidth);
+    const [clickedProfile, setClickedProfile] = useState(false);
+    const handleProfileBar = () => setClickedProfile(!clickedProfile);
+    const [filterDropDown, setFilterDropDown] = useState(false);
+    const hanleFilterDropDown = () => setFilterDropDown(!filterDropDown);
+
+    React.useEffect(() => {
+  
+      window.addEventListener("resize", () => setViewportWidth(window.innerWidth));
   
+  
+    }, []);
+    
+
+   
+  let pages : MenuPage[] = [];
+  let pages2 : MenuPage[] = []; 
+  
+
+
+  if (viewportWidth > 780) {
```

Comment:
This if can be a boolean from the Theme state of the app, so its a viewportMd or isMD.
have a look at context in react for this case.